--- ../../modelica2/modelicacc-code/util/ast_util.cpp	2014-06-11 10:17:19.133904282 -0300
+++ ast_util.cpp	2014-06-11 10:17:46.225905113 -0300
@@ -96,9 +96,12 @@
   case EXPCOMPREF:
   {
     AST_Expression_ComponentReference compRefA = a->getAsComponentReference();
+    AST_Expression_ComponentReference compRefB = b->getAsComponentReference();
     VarInfo varInfoA = _symbolTable->lookup(CREF_NAME(compRefA));
-    if (varInfoA != NULL && varInfoA->type()->getType() == TYARRAY) {
-      return compareArrays(compRefA, b->getAsComponentReference());
+    VarInfo varInfoB = _symbolTable->lookup(CREF_NAME(compRefB));
+    if ((varInfoA != NULL && varInfoA->type()->getType() == TYARRAY)
+			&& (varInfoB != NULL && varInfoB->type()->getType() == TYARRAY)) {
+      return compareArrays(compRefA, compRefB);
     } else {
       return (CREF_NAME(a).compare(CREF_NAME(b)) == 0);
     }
