--- ../modelica2/modelicacc-code/causalize/cycles_identification_strategy.h	2014-06-11 09:31:05.789819091 -0300
+++ causalize/cycles_identification_strategy.h	2014-06-09 11:19:06.221913579 -0300
@@ -5,7 +5,8 @@
  *      Author: fede
  */
 
-#include <causalize/graph_definition.h>
+//#include <causalize/graph_definition.h>
+#include <causalize/causalize2/graph_definition.h>
 
 #include <boost/config.hpp>
 #include <boost/graph/adjacency_list.hpp>
